conda_smithy/configure_feedstock.py

```diff
@@ -2685,13 +2907,13 @@ def set_migration_fns(forge_dir, forge_config):
 
 
 def main(
-    forge_file_directory,
-    forge_yml=None,
-    no_check_uptodate=False,
-    commit=False,
-    exclusive_config_file=None,
-    check=False,
-    temporary_directory=None,
+    forge_file_directory: str,
+    forge_yml: None = None,
```

Comment:
Is this always none?

Reply:
oops no, i updated it to be `Optional[str]` to have both str and None. (I use `Optional` instead of `|` because `|` is causing error for python 3.8.19, which is what github actions is running..)